### ISSUE TITLE
Add `prefer-validate` rule

### ISSUE THREAD

**Issue by ulrichstark** [marcalexiei/eslint-plugin-zod#426]
### 🧩 Description

The `validate` method was released in Zod 4.6: https://zod.dev/blog/zod-4-6?id=validate#validate

This proposed rule should flag calls of `safeParse()` where only `success` of the parse result is used and should suggest to replace it with the shorter and more performant call of `validate()`.

### 📦 Target plugin(s)

- [x] eslint-plugin-zod
- [x] eslint-plugin-zod-mini

### ⚙️ Options

-

### ✅ Valid Code Examples

```typescript
const UserSchema = z.object({
  name: z.string()
});

const success = UserSchema.validate(data);
```

### ❌ Invalid Code Examples

```typescript
const UserSchema = z.object({
  name: z.string()
});

const success = UserSchema.safeParse(data).success;
```

**Comment by marcalexiei** [comment]
Thanks for the report!

Would you be interested in submitting a PR yourself to add the new rule to both plugins?

**Comment by ulrichstark** [comment]
Yes, sure! You can assign me and I will find some time over the next days.

**Comment by marcalexiei** [comment]
Available in:
- [`eslint-plugin-zod@4.13.0`](https://github.com/marcalexiei/eslint-zod/releases/tag/eslint-plugin-zod%404.13.0)
- [`eslint-plugin-zod-mini@1.10.0`](https://github.com/marcalexiei/eslint-zod/releases/tag/eslint-plugin-zod-mini%401.10.0)
- [`eslint-plugin-zod-core@1.1.0`](https://github.com/marcalexiei/eslint-zod/releases/tag/eslint-plugin-zod-core%401.1.0)

**Comment by ulrichstark** [comment]
Thanks! Works as expected and found ten cases in our project to simplify.

The only issue (but not relevant to this project) is, that now in some cases `@typescript-eslint/no-unnecessary-condition` with enabled `checkTypePredicates` reports some of the new code like the following with the message "Unnecessary conditional, expression already has the type being checked by the type guard.": 

```ts
function check(input: string) {
  return z.email().validate(input);
}
```

I will just add `// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- Zod checks runtime refinements.` for these cases.

**Comment by marcalexiei** [comment]
I added a note in documentation of all 3 plugin rules. 
See #426.

**Comment by ulrichstark** [comment]
> I added a note in documentation of all 3 plugin rules. See [#426](https://github.com/marcalexiei/eslint-zod/issues/426).

Makes sense! You probably meant #445 :)

**Comment by marcalexiei** [comment]
Me after realising I posted the wrong link:

![Guy hit by a hale bale](https://media.tenor.com/9T0uxnjnS-UAAAAM/epic-embed-fail-epic-embed-failure.gif)
